eblocker-icapserver/src/main/java/org/eblocker/server/icap/resources/ResourceHandler.java

```diff
@@ -77,7 +78,7 @@ public static List<String> readLines(EblockerResource resource) {
             log.error("Error while reading all lines from this resource {}.", resource.getPath(), e);
         }
         // ignore
-        return null;
+        return new ArrayList<>();
```

Comment:
With lists, I prefer it when an empty list comes back and not a null. If I have looked at the code correctly, the calling methods can handle an empty list better than null. It's not that important, I just noticed it while looking through the code

Reply:
I think this would be a big change in behaviour. The methods calling `readLines()` do not check the return value as far as I can tell. So currently the calling code will crash with an NPE when it tries to use the list.

If we return an empty list the method fails silently (apart from the log message).

I think it would be better to throw an exception like in method `load()`. That would be more explicit and simplify debugging.

Reply:
In practice, the `IOUtils.readLines()` method will almost never throw an `IOException`. For example, if the resource is a file that does not exist, the call to `getFileInputStream()` above will throw an `EblockerException` which will not be caught.

So I simply used the same approach as in `ResourceHandler.load()`: log the error and throw an exception.